pkg/reconciler/v1alpha1/route/reconcile_resources.go

```diff
@@ -49,7 +49,7 @@ func (c *Reconciler) getClusterIngressForRoute(route *v1alpha1.Route) (*netv1alp
 		return nil, err
 	}
```

Comment:
since we now know the ingress instance name beforehand, we can simplify the logic here in `getClusterIngressForRoute` to look for the ingress instance directly

Reply:
yeah, backward compatibility would be an issue.
In fact, I have been entangled for long that whether we are considering this too early?
As we are still alpha, it should be the best time for us to focus more on functionality, once we bump to 1.x, things would be tough.

Reply:
I think focusing on upgrades early is a forcing function to keep things going smoothly.  /cc @mattmoor for his thoughts about this.